### PR TITLE
WIP: Embedded-Hal integration

### PR DESCRIPTION
**Pull request by Stupremee**
This PR implements some traits from the [embedded-hal](https://crates.io/crates/embedded-hal) crate.

### CODE REVIEW

File: src/hal/pwm.rs [hunk 1]
```diff
@@ -0,0 +1,27 @@
+use crate::pwm::PwmChannel;
+use embedded_hal::PwmPin;
+
+impl PwmPin for PwmChannel {
```
Comment (vbe0201) [inline src/hal/pwm.rs:4]: Be careful here, `PwmPin` requires the `hal-unproven` feature.

**Review by vbe0201 — COMMENTED** [review]
Looks good for most parts, see comments.

File: src/hal/pwm.rs (hunk 1 above)
Comment (Stupremee) [reply src/hal/pwm.rs:4]: `PwmPin` doesn't require the `hal-unproven` feature

Comment (vbe0201) [reply src/hal/pwm.rs:4]: After re-checking, it seems like only the `Pwm` trait which isn't being implemented here depends on this feature. Forget what I said, my bad.

**Pull request merged**